src/core_plugins/kibana/public/management/sections/indices/edit_index_pattern/scripted_fields_table/scripted_fields_table.js

```diff
@@ -61,34 +59,23 @@ export class ScriptedFieldsTable extends Component {
     this.setState({
       fields,
       deprecatedLangsInUse,
-      ...this.computeTableState(this.state.criteria, this.props, fields)
     });
   }
 
-  onDataCriteriaChange = criteria => {
-    this.setState(this.computeTableState(criteria));
-  }
+  getFilteredItems = () => {
+    const { fields } = this.state;
+    const { fieldFilter, scriptedFieldLanguageFilter } = this.props;
 
-  componentWillReceiveProps(nextProps) {
-    if (this.props.fieldFilter !== nextProps.fieldFilter) {
-      this.setState(this.computeTableState(this.state.criteria, nextProps));
+    if (fieldFilter) {
+      const normalizedFieldFilter = this.props.fieldFilter.toLowerCase();
+      return fields.filter(field => field.name.toLowerCase().includes(normalizedFieldFilter));
```

Comment:
We probably also need to leave in the `componentWillReceiveProps` and call `getFilteredItems` when either filter changes. We probably need to also refactor `getFilteredItems` to accept the filters or default to getting them off the prop list. We can't always do the latter because we're calling it from `componentWillReceiveProps` and `this.props` will not equal `nextProps` in that function.

See https://github.com/elastic/kibana/pull/16649/commits/119644584f551e0c6899ba553c3f7e262eb2f8d9#diff-3cebde7c25eb5abea1104b7f91a9f8bdR46

Reply:
After talking this through with @cjcenizal, I think I was promoting a poor pattern of storing computed data which makes it hard to manage the original data set as well as the computed set. It doesn't seem like a good idea to store them _both_ so I think we should move to a pattern where we only store the original data set and calculate the computed data during the `render()` method. To help on performance of potentially recalculating computed data unnecessarily, let's use [reselect](https://github.com/reactjs/reselect) which is a _tiny_ library designed to only call a function when the parameters have changed.

See https://github.com/elastic/kibana/pull/16649/commits/56e76d2c49353e1db1c914b4a7e5663142ec0907